src/util-thash.c

```diff
@@ -189,12 +189,16 @@ static void THashDataFree(THashTableContext *ctx, THashData *h)
     if (h != NULL) {
         DEBUG_VALIDATE_BUG_ON(SC_ATOMIC_GET(h->use_cnt) != 0);
 
+        uint32_t len = 0;
         if (h->data != NULL) {
+            if (ctx->config.DataLength) {
+                len = ctx->config.DataLength(h->data);
+            }
             ctx->config.DataFree(h->data);
         }
         SCMutexDestroy(&h->m);
         SCFree(h);
-        (void) SC_ATOMIC_SUB(ctx->memuse, THASH_DATA_SIZE(ctx));
+        (void)SC_ATOMIC_SUB(ctx->memuse, THASH_DATA_SIZE(ctx) + (uint64_t)len);
```

Comment:
is `THashDataFree` the only way we expire/discard data? `THashDataGetNew` code suggests there may be more paths to discarding a `THashData` w/o freeing it?

Reply:
> there may be more paths to discarding a THashData w/o freeing it

I am unable to understand that if there is a path like that, why doesn't it lead to a memleak?
I did find a path that I had missed but it involves freeing the data..